tests/integration/targets/activation/tasks/main.yml

```diff
@@ -0,0 +1,269 @@
+      register: tempdir
+
+    - name: Generate a local SSH key
+      local_action: "shell ssh-keygen -b 2048 -t rsa -f {{ tempdir.stdout }}/id_rsa -q -N 'passphrase'"
+
+    - name: Read the generated key
+      set_fact:
+        ssh_key_data: "{{ lookup('file', tempdir.stdout + '/id_rsa') }}"
+
+    - name: Create a new SCM credential
+      ansible.eda.credential:
+        <<: *credential_defaults
+        name: "{{ credential_name }}"
+        state: present
+        credential_type_name: Source Control
+        inputs:
+          username: joe
+          password: secret
+          ssh_key_data: "{{ ssh_key_data }}"
+          ssh_key_unlock: "passphrase"
+      register: credential_creation
+
+    - name: Assert that the credential was created successfully
+      assert:
+        that:
+          - credential_creation is changed
+          - credential_creation is success
+
+    - name: Create a new project
+      ansible.eda.project:
+        <<: *credential_defaults
+        name: "{{ project_name }}"
+        description: "Test Project Description"
+        url: "{{ scm_url }}"
+        credential: "{{ credential_name }}"
+        state: present
+      register: project_creation
+
+    - name: Assert that the project was created successfully
+      assert:
+        that:
+          - project_creation is changed
+          - project_creation is success
+
+    - name: Create a new decision environment
+      ansible.eda.decision_environment:
+        <<: *credential_defaults
+        name: "{{ decision_env_name }}"
+        description: "Test Decision Environment Description"
+        credential: "{{ credential_name }}"
+        image_url: "{{ image_url }}"
+      register: decision_environment_creation
+
+    - name: Create a new rulebook activation in check mode
+      ansible.eda.activation:
+        <<: *credential_defaults
+        name: "{{ activation_name }}"
+        description: "Example Activation description"
+        project_name: "{{ project_name }}"
+        rulebook_name: "{{ rulebook_name }}"
+        decision_environment_name: "{{ decision_env_name }}"
+        enabled: False
+        awx_token_name: "{{ awx_token_name }}"
+      check_mode: true
+      register: _result
+
+    - name: Check rulebook activation creation in check mode
+      assert:
+        that:
+          - _result.changed
+
+    - name: Create a new rulebook activation
+      ansible.eda.activation:
+        <<: *credential_defaults
+        name: "{{ activation_name }}"
+        description: "Example Activation description"
+        project_name: "{{ project_name }}"
+        rulebook_name: "{{ rulebook_name }}"
+        decision_environment_name: "{{ decision_env_name }}"
+        enabled: False
+        awx_token_name: "{{ awx_token_name }}"
+      register: _result
+
+    - name: Check rulebook activation creation
+      assert:
+        that:
+          - _result.changed
+
+    - name: Create a new rulebook activation again
```

Comment:
I getting an error Create a new rulebook activation again
`fatal: [localhost]: FAILED! => {"changed": false, "msg": "Failed to create/update rulebook activation: HTTP error {'detail': 'Method \"PATCH\" not allowed.'}"}`

Reply:
I added a note about the fact that the module does not support PATCH and also handled that at the module level. Let me know if that makes sense to you. Thanks.

Reply:
Makes sense, but I'm getting an error about the assert checking result message:

`[WARNING]: conditional statements should not include jinja2 templating delimiters such as {{ }} or {% %}.
Found: 'A rulebook activation with name: {{ activation_name }} already exists. The module does not support
modifying a rulebook activation.' in _result.msg
fatal: [localhost]: FAILED! => {"msg": "The conditional check ''A rulebook activation with name: {{ activation_name }} already exists. The module does not support modifying a rulebook activation.' in _result.msg' failed. The error was: Conditional is marked as unsafe, and cannot be evaluated."}`

Reply:
Removed jinja templating from the assert.